packages/test-utils/src/reporter.ts

```diff
@@ -242,6 +243,8 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
     new Map();
   private _testRunShardIdPromise: Promise<TestRunPendingWork> | null = null;
   private _uploadStatusThreshold: UploadStatusThresholdInternal = "none";
+  private _cacheUserIdPromise: Promise<void> | null = null;
```

Comment:
Given the restriction with "no async" in most of Reporter method, we should also have a random UUID that is persistent at a file path. That would give us some identifier to distinguish logs from one device to another.

Reply:
I agree, but that seemed out of scope for this ticket.

What do you think about a follow-up ticket for that work?

Reply:
Follow-up sounds good.

Reply:
https://linear.app/replay/issue/PRO-644/add-persistent-random-device-id-to-grafana-logs